java/src/jmri/jmrit/symbolicprog/SplitHundredsVariableValue.java

```diff
@@ -0,0 +1,92 @@
+package jmri.jmrit.symbolicprog;
+
+import java.util.HashMap;
+import javax.swing.JLabel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Like {@link SplitVariableValue}, except that the string representation is split out
+ * into 100's in each CV (up to two in this first implementation)
+ * <br><br>
+ * All the attributes of {@link SplitVariableValue} are inherited.
+ *
+ *
+ * @author Bob Jacobsen Copyright (C) 2001, 2002, 2003, 2004, 2013, 2014
+ * @author Dave Heap Copyright (C) 2016
+ */
+public class SplitHundredsVariableValue extends SplitVariableValue {
+
+    public SplitHundredsVariableValue(String name, String comment, String cvName,
+            boolean readOnly, boolean infoOnly, boolean writeOnly, boolean opsOnly,
+            String cvNum, String mask, int minVal, int maxVal,
+            HashMap<String, CvValue> v, JLabel status, String stdname,
+            String pSecondCV, int pFactor, int pOffset, String uppermask, String extra1, String extra2, String extra3, String extra4) {
+        super(name, comment, cvName, readOnly, infoOnly, writeOnly, opsOnly, cvNum, mask, minVal, maxVal, v, status, stdname, pSecondCV, pFactor, pOffset, uppermask, extra1, extra2, extra3, extra4);
+    }
+
+    @Override
+    public void stepOneActions(String name, String comment, String cvName,
+            boolean readOnly, boolean infoOnly, boolean writeOnly, boolean opsOnly,
+            String cvNum, String mask, int minVal, int maxVal,
+            HashMap<String, CvValue> v, JLabel status, String stdname,
+            String pSecondCV, int pFactor, int pOffset, String uppermask, String extra1, String extra2, String extra3, String extra4) {
+
+        if (extra3 != null) {
+            _minVal = getValueFromText(extra3);
+        }
+        if (extra4 != null) {
+            _maxVal = getValueFromText(extra4);
+        }
+    }
+
+    @Override
+    long getValueFromText(String s) {
+        if (s.isEmpty()) return 0L;
+        long val = Long.parseUnsignedLong(s);
+        long result = 0;
+        long multiplier = 1L;
+        final long INRADIX = 100L;
+        final long OUTRADIX = 0x100;
+        while (val > 0) {
+            long digits = val % INRADIX;
+            val = val/INRADIX;
+            result = result + multiplier*digits;
+            multiplier = multiplier * OUTRADIX;
+        }
+        return result;
+    }
+
+    @Override
+    String getTextFromValue(long val) {
+        String result = "";
+
+        final long INRADIX = 0x100;
+        while (val > 0) {
+            long digits = val % INRADIX;
+            val = val/INRADIX;
+            result = "" + digits + result;// that's a String prepend operation
```

Comment:
I'm reading this on my phone so I might be wrong. But it seems to me that the value 0x0102 would result in the string "21" which would be wrong in two ways:
- The digits are in the wrong order. "21" instead of "12".
- The digits aren't padded with zero. "21" instead of "201".

But maybe I don't understand what this method is supposed to do. I expected the result to be "102" or "0102".

Reply:
Thanks!  You're right, it's not handling leading zeros correctly.  (The order was right, as it's given by the underlying CV order)